### ISSUE TITLE
Importing ABCs from collections directly is deprecated

### ISSUE THREAD

**Issue by stuartarchibald** [numba/numba#3251]
```
PYTHONWARNINGS=always python -c 'from numba import jit'
```
shows a few deprecation warnings for `collections` under Python 3.7, essentially:
```
DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
```
these clearly need fixing ahead of Python 3.8.

**Comment by azrdev** [comment]
#3256 apparently didn't catch all occurences: With python-numba 0.43.1-1 on archlinux I still get 
~~~
  /usr/lib/python3.7/site-packages/numba/typed/typeddict.py:4: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import MutableMapping
~~~

Searching here on github reveals even more instances: <https://github.com/numba/numba/search?q=%22from+collections+import%22>

**Comment by stuartarchibald** [comment]
@azrdev thanks, https://github.com/numba/numba/pull/3935 fixed this for 0.44, the release candidate for which should be out imminently.

Also, from a checkout of master branch:
```
$ grep -r "from collections import" numba|cut -d ':' -f 2|sed -e 's/^\ *//g'|sort|uniq
from collections import defaultdict
from collections import defaultdict, deque
from collections import defaultdict, namedtuple
from collections import defaultdict, OrderedDict
from collections import defaultdict, OrderedDict, namedtuple
from collections import deque
from collections import Mapping, MutableMapping, Sequence
from collections import namedtuple
from collections import namedtuple, defaultdict
from collections import namedtuple, deque
from collections import namedtuple, OrderedDict
from collections import OrderedDict

$ grep -r "Mapping, MutableMapping, Sequence" numba
numba/six.py:    from collections import Mapping, MutableMapping, Sequence
numba/six.py:    from collections.abc import Mapping, MutableMapping, Sequence
```
seems like it's ok?

**Comment by azrdev** [comment]
@stuartarchibald thx, didn't find that one. looks like next release fixes the issue, yes.
